Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java

```diff
@@ -105,10 +116,57 @@ public boolean onQueryTextChange(String newText) {
                 return true;
             }
         });
-        fetchCitiesList();
+
+        mDatabase = AppDataBase.getAppDatabase(getActivity());
+        List<City> mCities = Arrays.asList(mDatabase.cityDao().loadAll());
+        if (checkCachedCities(mCities)) {
+            fetchCitiesList();
+        } else {
+            FlipSettings settings = new FlipSettings.Builder().defaultPage().build();
+            lv.setAdapter(new CityAdapter(mActivity, mCities, settings));
+            lv.setOnItemClickListener((parent, mView, position, id1) -> {
+                City city = (City) lv.getAdapter().getItem(position);
+                Intent intent = FinalCityInfoActivity.getStartIntent(mActivity, city);
+                startActivity(intent);
+            });
+        }
         return view;
     }
 
+    /*
+     * Check cached cities with expiration time 24 hours
```

Comment:
Add javadoc style comments to the newly added funtions (that's what we are following)
It explains about parameters & return values too